File: src/main/java/info/idgst/digest/reader/DigestReader.java
package info.idgst.digest.reader;

import info.idgst.digest.Digest;

/**
 * Service foe reading {@link Digest}s form different files, like JSON, xml, etc.
 *
 * @author Aliaksei Bahdanau
 */
public interface DigestReader {

    /**
     * Converts specified bytes to {@link Digest}.
     *
     * @param bytes
     * @return {@link Digest}
     */
    Digest readDigest(byte[] bytes);
}

File: src/main/kotlin/info/devbug/digest/util/DigestReader.kt
package info.devbug.digest.util

import info.devbug.digest.repository.DigestDto

/**
 * @author Aliaksei Bahdanau
 */
interface DigestReader {

    /**
     * Reads the Digest from the specified file.
     */
    fun readDigest(filePath: String): DigestDto
}